src/cdk/table/table.ts

```diff
@@ -785,20 +790,26 @@ export class CdkTable<T> implements AfterContentChecked, CollectionViewer, OnDes
    */
   private _renderUpdatedColumns() {
     const columnsDiffReducer = (acc: boolean, def: BaseRowDef) => acc || !!def.getColumnsDiff();
+    let forced = false;
```

Comment:
It may be easier to read if we store the results of the reducers. Then we don't need to try to explain the `forced` variable, it'll be clear the condition we are returning, e.g.:

```

    // Force re-render data rows if the list of column definitions have changed.
    const dataColumnsChanged = this._rowDefs.reduce(columnsDiffReducer, false);
    if (dataRowDefsChanged) {
      this._forceRenderDataRows();
    }

    // Force re-render header/footer rows if the list of column definitions have changed.
    const headerColumnsChanged = this._rowDefs.reduce(columnsDiffReducer, false);
    if (this._headerRowDefs.reduce(columnsDiffReducer, false)) {
      this._forceRenderHeaderRows();
    }

    const footerColumnsChanged = this._rowDefs.reduce(columnsDiffReducer, false);
    if (this._footerRowDefs.reduce(columnsDiffReducer, false)) {
      this._forceRenderFooterRows();
    }
    
    return dataColumnsChanged || headerColumnsChanged || footerColumnsChanged;
```